pyam/core.py

```diff
@@ -1361,11 +1378,31 @@ def _meta_idx(data):
     return data[META_IDX].drop_duplicates().set_index(META_IDX).index
 
 
-def _aggregate(df, by):
+def _aggregate(df, by, aggregator='sum', weightvar_df=None):
     """Aggregate `df` by specified column(s), return indexed `pd.Series`"""
     by = [by] if isstr(by) else by
     cols = [c for c in list(df.columns) if c not in ['value'] + by]
-    return df.groupby(cols).sum()['value']
+    # pick aggregator func (default: sum)
+    if aggregator == 'w.avg' and not weightvar_df.empty:
+        calcdf = pd.merge(df, weightvar_df,
+                          how='left', suffixes=('', '_weight'),
```

Comment:
just out of curiosity - why `left`? is it similar to SQL's join type? what witl it do with records from left frame having no corresponding records in right one?

Reply:
good question. we could use it to generate error messages about missing weights. but I'm not sure whether that will make a difference. 
@danielhuppmann any hints how to deal with partially missing weights?